README.md
<img src="https://user-images.githubusercontent.com/63138398/225706659-4fa396f3-fc8a-4cdc-bbd4-3600b323db97.svg" alt="Button Preview" > 

  <h1 align="left">neubrutalism_ui v2.0.0 🥳</h1>
<p align="center">
<a href="https://github.com/deepraj02/neubrutalism_ui/issues">Request Feature</a>
<br>
<a href="https://github.com/deepraj02/neubrutalism_ui/issues">Report Bug</a>
</p>

![Contributors](https://img.shields.io/github/contributors/deepraj02/neubrutalism_ui?color=dark-green) ![Stargazers](https://img.shields.io/github/stars/deepraj02/neubrutalism_ui?style=social) ![Issues](https://img.shields.io/github/issues/deepraj02/neubrutalism_ui) ![License](https://img.shields.io/github/license/deepraj02/neubrutalism_ui) 

## Table Of Contents

- [](#)
      - [Examples:](#examples)
  - [Roadmap](#roadmap)
  - [Contributing](#contributing)
    - [Creating A Pull Request](#creating-a-pull-request)
  - [License](#license)
  - [Authors](#authors)

## About The Project

"Neubrutalism UI is a comprehensive UI kit for Flutter that incorporates the principles of Neubrutalism design. With this package, you can easily add beautiful and functional UI components to your Flutter app in a style that emphasizes simplicity and minimalism.

The package includes a variety of UI components such as buttons, cards, forms, and more, each designed with a subtle shadow and rounded corners that add depth and texture to your app. You can customize the appearance of each component to fit the specific needs of your app, including changing the border radius, depth, and color.

Whether you're a beginner or an experienced developer, Neubrutalism UI makes it easy to create elegant and functional UI components in Flutter that align with the principles of Neubrutalism design. With clear and concise documentation and easy installation instructions, you can start using the package in your app today."

## Built With [Flutter](https://flutter.dev/) 💙

## Getting Started
<!-- Whats the differnect v1.0.0 vs 2.0.0 -->

<!-- TODO: Replace v 0.2.0 with this.
v2.0.0 
NeuTextButton(
                  text: const Text(
                    "Hello ",
                    style: TextStyle(
                      fontSize: 40,
                    ),
                  ),

1 . Replaced Widget `child` with `text` (Text) {Breaking Change}.

1. Fixed the Initial UI of the Widget. (Predefined some properties which  are customizeable).

 -->
# What's the Difference !!! `v1.0.0` vs `v2.0.0`✨

Includes Breaking Changes (Refer to Changelog),

<table>
<tr>
<td>

```dart
NeuTextButton(
  borderColor: Colors.black,
  shadowColor: Colors.black,
  buttonHeight: 50,
  buttonWidth: MediaQuery.of(context).size.width * 0.5,
  child: Row(
    mainAxisAlignment: MainAxisAlignment.center,
    children: const [
      Padding(
        padding: EdgeInsets.all(8.0),
        child: Text(
          "Hello",
          style: TextStyle(
              fontSize: 20, fontWeight: FontWeight.w600),
        ),
      ),
    ],
   ),
 ),
```
</td>
<td>

<!-- V 1.0.0 -->
```dart
NeuTextButton(
      borderRadius: BorderRadius.circular(12),
      buttonColor:
          const Color.fromARGB(255, 236, 199, 211),
      buttonHeight: 60,
      buttonWidth: 100,
      enableAnimation: true,
      text: Text(
        "Hello",
        style: GoogleFonts.robotoCondensed(
            textStyle: const TextStyle(
          fontSize: 20,
          fontWeight: FontWeight.w700,
        ),
        ),
      ),
),
```
</td>
</tr>
</table>

# Usage

### NeuTextButton with Customization (v2.0.0) : Enable Cool Animation
#### (Set `enableAnimation=true`)
Animations are currently mapped to the value fo offset provided in the `offset` property in the `NeuTextButton` and `NeuIconButton` widgets. 
(P.S: More the Value of the offset more will be the onPressed bounce animation Value.)
<table>
<tr>
<td>


```dart
NeuTextButton(
      borderRadius: BorderRadius.circular(12),
      buttonColor:
          const Color.fromARGB(255, 236, 199, 211),
      buttonHeight: 60,
      buttonWidth: 100,
      enableAnimation: true,
      text: Text(
        "Hello",
        style: GoogleFonts.robotoCondensed(
            textStyle: const TextStyle(
          fontSize: 20,
          fontWeight: FontWeight.w700,
        ),
        ),
      ),
),
```
</td>
<td>
<img  src="https://user-images.githubusercontent.com/63138398/265234887-341632d2-5b7c-4e09-b226-7982e4de86c6.gif"  alt="">
</td>
</tr>
</table>


### The Neu - NeuBottomNav 

#### Usage ans Setup

In-order to disable the container overflow behind the Bottom Nav we need to specify the following flags in both the `Scaffold` and `SafeArea` widget.

The NeuBottomNav contains some additional flags such as `hideOnScroll`, `enableFloating` etc:
```dart
Scaffold(
      resizeToAvoidBottomInset: false,
      extendBody: true,
      child:SafeArea(
        bottom: false,
      ))
```

#### Examples:

<img  src="https://user-images.githubusercontent.com/63138398/265235364-a0afb9b4-1bb7-437b-8cd1-795483e0d781.png"  alt="">

## Roadmap

See the [open issues](https://github.com/deepraj02/neubrutalism_ui/issues) for a list of proposed features (and known issues).

## Contributing

Contributions are what make the open source community such an amazing place to be learn, inspire, and create. Any contributions you make are **greatly appreciated**.
* If you have suggestions for adding or removing projects, feel free to [open an issue](https://github.com/deepraj02/neubrutalism_ui/issues/new) to discuss it, or directly create a pull request after you edit the *README.md* file with necessary changes.
* Please make sure you check your spelling and grammar.
* Create individual PR for each suggestion.


### Creating A Pull Request

- Fork the Project
- Create your Feature Branch (`git checkout -b feature/AmazingFeature`)
- Commit your Changes (`git commit -m 'Add some AmazingFeature'`)
- Push to the Branch (`git push origin feature/AmazingFeature`)
- Open a Pull Request (`choose the Staging branch From the Dropdown`)

## License

Distributed under the MIT License. See [LICENSE](https://github.com/deepraj02/neubrutalism_ui/blob/master/LICENSE.md) for more information.

## Author
* **Deepraj Baidya** - *Flutter SDE Intern* - [Deepraj Baidya](https://github.com/deepraj02) - *🤔*

## Contributors
![Contributors](https://contrib.rocks/image?repo=deepraj02/neubrutalism_ui)

CHANGELOG.md
# 2.0.2 (10-11-2023)

## Release Highlights
Performance and Bug Fixes.

### Changes
- [x] fix:  Reversed animation and onTap as seperate concepts in InkWell widget

### Closes 
- [x] closes #45

# 2.0.1 (09-11-2023)

## Release Highlights
Performance and Bug Fixes.

### Changes
- [x] `SearchBarIcon` is renamed as `leadingIcon` for `NeuSearchBar` Widget. (Done to make the properties more self-Explainatory).
- [x] fix: Stationary Shadow in `NeuButtons` (The offset staying stationary onPressing).
- [x] fix:  `OnPressed` function not invoking while `enableAnimation` value set to true.

### Closes 
- [x] closes #39
- [x] closes #41
- [x] closes #40



# 2.0.0 (26-10-2023)

## Release Highlights
2.0.0 (Feature Update) will focused on adding more UI components and will reduce the complexity of creating Custom UI from this Package.
### Breaking Changes
- [x] `NeuCard` is depriciated now and will be Removed in the Following Versions. (For better Styling Freedom use `NeuContainer` instead).
- [x] `NeuTextButton` contains some breaking changes as the `Widget` child properties is removed completely and replaced with `Text` Property focusing only on keeping Texts in the `NeuTextButton`. (For Complex UI design choose `NeuContainer`)
- [x] Removed the property `paddingData` from `NeuIconButton`
### Changes
- [x] Migrated to NeuContainer instead of Material Container in all widgets of this Package `NeuTextButton`  `NeuSearchBar` & `NeuIconButton`.
- [x] Changed the Initial UI Lookup Styling for widgets like `NeuTextButton` & `NeuSearchBar`.
### Features
- [x] OnPressed animation added in v2.0.0 for `NeuTextButton` & `NeuIconButton` (accessable by the `enableAniamtion` and `animationDuration` flags).

### Documentation
- [x] `example/main.dart` revamp to contain all NeuWidgets.
- [ ] Code Snippets and Set-up process explained for all the widgets

### Issues Fixed
- [x] closes #28 
- [x] closes #24 
- [x] closes #29 
- [x] closes #28 
- [x] closes #25 

### TODO's
- [ ] closes #26 
- [ ] closes #33 

# 2.0.0-beta-2(03-09-2023)

## Release Highlights

- [x] Docs Update for v2.0.0-beta()
- [x] Minor UI and Code Fixes
- [x] Example app revamp to include all Widgets.

# 2.0.0-beta (22-08-2023)

## Release Highlights
2.0.0 (Feature Update) will focused on adding more UI components and will reduce the complexity of creating Custom UI from this Package.
New Widgets  `NeuBottomNav` will also be included along with other bug fixes and Performance improvements and Improved code for better testability
### Breaking Changes
- [x] `NeuCard` is depriciated now and will be Removed in the Following Versions. (For better Styling Freedom use `NeuContainer` instead).
- [x] `NeuTextButton` contains some breaking changes as the `Widget` child properties is removed completely and replaced with `Text` Property focusing only on keeping Texts in the `NeuTextButton`. (For Complex UI design choose `NeuContainer`).
- [x] `NeuBottomNav` requires adding some additional `Scaffold` and `SafeArea` flags as of now. 
- [x] Removed the property `paddingData` from `NeuIconButton`
### Changes
- [x] Migrated to NeuContainer instead of Material Container in all widgets of this Package `NeuTextButton`  `NeuSearchBar` & `NeuIconButton`.
- [x] Changed the Initial UI Lookup Styling for widgets like `NeuTextButton` & `NeuSearchBar`.
### Features
- [x]  `NeuBottomNav` -UI styling done and dusted (Navigation Part Under Development RN).
- [x] OnPressed animation added in v2.0.0 for `NeuTextButton` & `NeuIconButton` (accessable by the `enableAniamtion` and `animationDuration` flags).

### Documentation
- [x] `example/main.dart` revamp to contain all NeuWidgets.
- [ ] Code Snippets and Set-up process explained for all the widgets

### Issues Fixed
- [x] closes #28 
- [x] closes #24 
- [x] closes #29 
- [x] closes #28 
- [x] closes #25 
- [ ] closes #26 
- [ ] closes #33 


# 1.0.0 (15-04-2023)

## Release Highlights
### (Contains Breaking Changes)
- [x] docs: Update README to the Current Standard. 
- [x] docs: Add Updated API Documentation for All NeuWidgets.
- [x] feat: add textalign property in neuTextButton.
- [x] feat: Add a Container for Creating Custom Widgets in Neubrutalist style
- [x] feat: add nested widget support to provide more flexibility
- [ ] feat: add OnHover Animation
- [ ] feat: Add Widget Tests
- Issues Closed
  - [x] closes #12 
  - [x] closes #15 - Added Nested widget Support 
  - [x] closes #14 - Added NeuContainer
  - [x] closes #17  -  `shadowGeometry` -> `blurGeometry` 
  - [x]closes #13 - Added Fresh Docs 

# 0.2.0 (19-03-2023)
## Release Highlights:
- [x] Added 2 New Widgets `neuIconButton` and `neuSearchBar`
- [x] Fixed the bug causing no response on onPressed function in `neuTextButton`
 

# 0.1.2 (17-03-2023)
## Release Highlights:
- [x] Fix Pubspec


# 0.1.1 (17-03-2023)
## Release Highlights:
- [x] Added 2 amazing Widgets
- [x] Added More Documentation.
- [x] Added Example Section.
- [x] Fixes for NeuImageButton.


example/lib/main.dart
import 'package:flutter/material.dart';
import 'package:google_fonts/google_fonts.dart';
import 'package:neubrutalism_ui/neubrutalism_ui.dart';

void main() {
  runApp(const NeuBrutalism());
}

class NeuBrutalism extends StatelessWidget {
  const NeuBrutalism({super.key});

  @override
  Widget build(BuildContext context) {
    return const MaterialApp(
      debugShowCheckedModeBanner: false,
      home: ScreenWidget(),
    );
  }
}

class ScreenWidget extends StatelessWidget {
  const ScreenWidget({
    super.key,
  });

  @override
  Widget build(BuildContext context) {
    final controller = TextEditingController();
    return Scaffold(
      resizeToAvoidBottomInset: false,
      extendBody: true,
      // backgroundColor: const Color.fromARGB(255, 232, 210, 236),
      body: SafeArea(
          bottom: false,
          minimum: const EdgeInsets.only(top: 30, left: 10, right: 10),
          child: SingleChildScrollView(
            child: Column(
              children: [
                Row(
                  mainAxisAlignment: MainAxisAlignment.spaceBetween,
                  children: [
                    Text(
                      "Hi, Deepraj",
                      style: GoogleFonts.anton(
                        textStyle: const TextStyle(
                          fontSize: 30,
                        ),
                      ),
                    ),
                    NeuContainer(
                        color: Colors.blueAccent,
                        height: 50,
                        width: 50,
                        borderRadius: BorderRadius.circular(12),
                        offset: const Offset(3, 3),
                        child: const Icon(
                          Icons.supervised_user_circle_outlined,
                          size: 40,
                        )),
                  ],
                ),
                Align(
                  alignment: Alignment.topLeft,
                  child: Text(
                    "Become a Better Dev Today",
                    style: GoogleFonts.robotoCondensed(
                        textStyle: const TextStyle(
                      fontSize: 20,
                      fontWeight: FontWeight.w300,
                    )),
                  ),
                ),
                const SizedBox(
                  height: 30,
                ),
                Row(
                  mainAxisAlignment: MainAxisAlignment.spaceAround,
                  children: [
                    NeuSearchBar(
                      searchController: controller,
                      keyboardType: TextInputType.name,
                      searchBarColor: Colors.white,
                      hintText: "What are you looking for",
                      borderRadius: BorderRadius.circular(12),
                    ),
                    NeuIconButton(
                      buttonWidth: 60,
                      buttonColor: const Color.fromARGB(255, 234, 203, 214),
                      borderRadius: BorderRadius.circular(10),
                      icon: const Icon(Icons.sort),
                      //onPressed: () => print("hii"),
                      enableAnimation: false,
                    )
                  ],
                ),
                const SizedBox(
                  height: 30,
                ),
                Align(
                  alignment: Alignment.centerLeft,
                  child: Text(
                    "Course in Progress",
                    style: GoogleFonts.robotoCondensed(
                        textStyle: const TextStyle(
                      fontSize: 20,
                      fontWeight: FontWeight.w700,
                    )),
                  ),
                ),
                const SizedBox(
                  height: 15,
                ),
                NeuContainer(
                  color: Colors.white,
                  height: 150,
                  width: double.maxFinite,
                  borderRadius: BorderRadius.circular(12),
                  borderWidth: 2,
                  child: Padding(
                    padding: const EdgeInsets.all(15.0),
                    child: Column(
                      children: [
                        const Spacer(),
                        Align(
                          alignment: Alignment.bottomRight,
                          child: NeuTextButton(
                            borderRadius: BorderRadius.circular(12),
                            buttonColor:
                                const Color.fromARGB(255, 236, 199, 211),
                            buttonHeight: 60,
                            buttonWidth: 100,
                            enableAnimation: true,
                            onPressed: () => debugPrint("hello"),
                            text: Text(
                              "Start",
                              style: GoogleFonts.robotoCondensed(
                                  textStyle: const TextStyle(
                                fontSize: 20,
                                fontWeight: FontWeight.w700,
                              )),
                            ),
                          ),
                        ),
                      ],
                    ),
                  ),
                ),
                const SizedBox(
                  height: 15,
                ),
                Align(
                  alignment: Alignment.centerLeft,
                  child: Text(
                    "Popular Course",
                    style: GoogleFonts.robotoCondensed(
                        textStyle: const TextStyle(
                      fontSize: 20,
                      fontWeight: FontWeight.w700,
                    )),
                  ),
                ),
                const SizedBox(
                  height: 15,
                ),
                SingleChildScrollView(
                  scrollDirection: Axis.horizontal,
                  child: Row(
                    children: [
                      NewWidget(
                        image: Image.network(
                            'https://images.unsplash.com/photo-1533518463841-d62e1fc91373?ixlib=rb-4.0.3&ixid=M3wxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8fA%3D%3D&auto=format&fit=crop&w=2940&q=80'),
                        text: 'Dexter Lab 101',
                      ),
                      const SizedBox(
                        width: 15,
                      ),
                      NewWidget(
                        image: Image.network(
                            'https://images.unsplash.com/photo-1640499900704-b00dd6a1103a?ixlib=rb-4.0.3&ixid=M3wxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8fA%3D%3D&auto=format&fit=crop&w=2942&q=80'),
                        text: 'Cosmos',
                      ),
                    ],
                  ),
                ),
                const SizedBox(
                  height: 12,
                ),
                Align(
                  alignment: Alignment.centerLeft,
                  child: Text(
                    "Top Mentors",
                    style: GoogleFonts.robotoCondensed(
                        textStyle: const TextStyle(
                      fontSize: 20,
                      fontWeight: FontWeight.w700,
                    )),
                  ),
                ),
                const SizedBox(
                  height: 12,
                ),
                const SingleChildScrollView(
                  scrollDirection: Axis.horizontal,
                  child: Row(
                    mainAxisSize: MainAxisSize.max,
                    children: [
                      MentorsCard(),
                      MentorsCard(),
                      MentorsCard(),
                      MentorsCard(),
                      MentorsCard(),
                      MentorsCard(),
                      MentorsCard(),
                    ],
                  ),
                )
              ],
            ),
          )),
    );
  }
}

class MentorsCard extends StatelessWidget {
  const MentorsCard({
    super.key,
  });

  @override
  Widget build(BuildContext context) {
    return NeuContainer(
      shadowColor: Colors.transparent,
      color: Colors.white,
      height: 80,
      width: 250,
      borderRadius: BorderRadius.circular(10),
      child: Row(
        children: [
          Padding(
            padding: const EdgeInsets.all(4.0),
            child: NeuCard(
              borderRadius: BorderRadius.circular(50),
              shadowColor: Colors.transparent,
              child: ClipRRect(
                borderRadius: BorderRadius.circular(50),
                child: Image.network(
                  'https://images.unsplash.com/photo-1534528741775-53994a69daeb?ixlib=rb-4.0.3&ixid=M3wxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8fA%3D%3D&auto=format&fit=crop&w=2864&q=80',
                ),
              ),
            ),
          ),
          Column(
            mainAxisAlignment: MainAxisAlignment.spaceAround,
            children: [
              Text(
                "John Doe",
                textAlign: TextAlign.start,
                style: GoogleFonts.robotoCondensed(
                  textStyle: const TextStyle(
                    fontSize: 26,
                    fontWeight: FontWeight.bold,
                  ),
                ),
              ),
              Text(
                "Full-Stack Developer",
                textAlign: TextAlign.start,
                style: GoogleFonts.robotoCondensed(
                  textStyle: const TextStyle(
                    fontSize: 20,
                  ),
                ),
              ),
            ],
          )
        ],
      ),
    );
  }
}

class NewWidget extends StatelessWidget {
  const NewWidget({
    super.key,
    required this.image,
    required this.text,
  });
  final Image image;
  final String text;

  @override
  Widget build(BuildContext context) {
    return Padding(
      padding: const EdgeInsets.all(8.0),
      child: NeuContainer(
        height: 260,
        width: 230,
        color: Colors.white,
        borderRadius: BorderRadius.circular(13),
        child: Column(
          children: [
            image,
            Padding(
              padding: const EdgeInsets.all(8.0),
              child: Align(
                alignment: Alignment.topLeft,
                child: Text(
                  text,
                  style: GoogleFonts.robotoCondensed(
                    textStyle: const TextStyle(
                      fontSize: 20,
                    ),
                  ),
                ),
              ),
            ),
            Padding(
              padding: const EdgeInsets.only(left: 5.0),
              child: Row(
                children: [
                  const Icon(Icons.access_time_outlined),
                  Text(
                    "30 hrs",
                    style: GoogleFonts.robotoCondensed(
                        textStyle: const TextStyle(
                      fontSize: 15,
                      fontWeight: FontWeight.w700,
                    )),
                  ),
                  const SizedBox(
                    width: 30,
                  ),
                  const Icon(Icons.star_border),
                  Text(
                    "4.5",
                    style: GoogleFonts.robotoCondensed(
                        textStyle: const TextStyle(
                      fontSize: 15,
                      fontWeight: FontWeight.w700,
                    )),
                  ),
                ],
              ),
            ),
            Padding(
              padding: const EdgeInsets.only(right: 12.0),
              child: Align(
                alignment: Alignment.bottomRight,
                child: NeuIconButton(
                    buttonHeight: 30,
                    buttonWidth: 30,
                    buttonColor: const Color.fromARGB(255, 248, 250, 194),
                    icon: const Icon(Icons.arrow_forward),
                    enableAnimation: true,
                    borderRadius: BorderRadius.circular(20)),
              ),
            ),
          ],
        ),
      ),
    );
  }
}
